Fix SYS COLUMNS schema in ODBC mode

bpintea opened the pull request:
This PR fixes a regression when certain ODBC-specific columns that need to
be of the short type were returned as the integer type.

Fixes #59506.

bpintea committed:
Fix SYS COLUMNS schema in ODBC mode

This fixes a regression when certain ODBC-specific columns that need to
be of the short type were returned as the integer type.

elasticmachine commented:
Pinging @elastic/es-ql (:Query Languages/SQL)

matriv commented on x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plan/logical/command/sys/SysColumnsTests.java:
```diff
@@ -501,10 +506,29 @@ public void testSysColumnsWithNullCatalog() throws Exception {
         }, mapping);
     }
 
+    private static void checkOdbcShortTypes(SchemaRowSet r) {
```
Personal preference: I like to have the private methods at the bottom of the class and not before they are first used.

bpintea replied:
Thanks, updated.
I've also updated the actual fix, since it wasn't entirely correct regarding pagination.

matriv reviewed (approved):
LGTM. Left a minor nit pick comment.

bpintea committed:
Fix results pagination

A ListCursor needs to be providing the Page, otherwise the pagination is
not respected.

Also fix review comments.

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into fix/59506

matriv reviewed (commented):
Can we have a test with small page size like 1,2 or 3 ?

bpintea committed:
Fix test stubbing. Add cursor test

This fixes the stubbing for the *-indices SYS COLUMN commands.

It also adds testing for response pagination in ODBC mode.

bpintea committed:
Fix style violation

Trim a line to 140 border.

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into fix/59506

bpintea commented:
> Can we have a test with small page size like 1,2 or 3 ?

Yes, good idea. Figured out and fixed a different test stubbing mistake along it.

bpintea commented:
@elasticmachine run elasticsearch-ci/packaging-sample-windows

astefan reviewed (approved):
LGTM
